src/couch/src/couch_db.erl

```diff
@@ -1589,7 +1476,7 @@ open_doc_int(Db, #full_doc_info{id=Id,rev_tree=RevTree}=FullDocInfo, Options) ->
         {ok, Doc#doc{meta=doc_meta_info(DocInfo, RevTree, Options)}}, Options);
 open_doc_int(Db, Id, Options) ->
     case get_full_doc_info(Db, Id) of
-    {ok, FullDocInfo} ->
+    #full_doc_info{} = FullDocInfo ->
```

Comment:
Same as above: if `not_found` will be a first match, we can trust `get_full_doc_info/2` to do the right thing and drop the guard, seems less defensive. Though I don't feel too strong about it, either way is fine with me.